src/cueUtilities.js

```diff
@@ -268,6 +272,96 @@ module.exports = function (params, cy, api) {
+              x: (node.position("x") + node.width() / 2 + parseFloat(node.css('padding-right'))),
+              y: (node.position("y") + node.height() / 2+ parseFloat(node.css('padding-bottom')))};
+    
+            if (currMousePos.x >= topLeft.x && currMousePos.y >= topLeft.y &&
+              currMousePos.x <= bottomRight.x && currMousePos.y <= bottomRight.y){
+              return true;
+            }
+          }
+          return false;
+        }
+
+        function getAllGroups(cy) {
+          return cy.$('.cy-expand-collapse-collapsed-node').union(':parent');
+        }
+
+        function getAllGroupsUnderEvent(cy, e) {
+          const groupsUnderEvent = [];
+          const groups = getAllGroups(cy);
+          for (let i = 0; i < groups.length; i++) {
+            if (isInsideCompound(groups[i], e)) {
+              groupsUnderEvent.push(groups[i]);
+            }
+          }
+          return groupsUnderEvent;
+        }
+
+        function getAllGroupsWithHighestZDepth(groups) {
+          let highestZDepthGroups = [];
+          let highestZDepth = 0;
+          groups.forEach((group) => {
+            let currentZDepth = group.zDepth();
+            // if we find groups with a higher zDepth, keep track of those
+            if (currentZDepth > highestZDepth) {
+              highestZDepth = currentZDepth;
+              highestZDepthGroups = [group];
+            } else if (currentZDepth === highestZDepth) {
+              highestZDepthGroups.push(group);
+            }
+          });
+          return highestZDepthGroups;
+        }
+
+        function drawCueOnHighestZDepthGroup(e) {
+          const prevHoveredGroup = hoveredGroup;
+          const groupsUnderEvent = getAllGroupsUnderEvent(cy, e);
+          const highestZDepthGroups = getAllGroupsWithHighestZDepth(groupsUnderEvent);
+          if (highestZDepthGroups.length > 0) {
+            // take the last one as, we THINK cytoscape puts most recently created elements last in selecters
+            // the last element will be the one to select if we click on it when overlapping with other elements
+            hoveredGroup = highestZDepthGroups[highestZDepthGroups.length - 1];
+          } else {
+            hoveredGroup = null;
+          }
+          // to save computation, only redraw if we have a new hovered group
+          if ((prevHoveredGroup && !hoveredGroup) || !prevHoveredGroup || prevHoveredGroup.id() !== hoveredGroup.id()) {
+            refreshCanvasImages();
+          }
+        }
+
+        cy.on('mousemove', 'edge', data.eMouseMove = function(e) {
+          drawCueOnHighestZDepthGroup(e);
+        });
+
+        cy.on('mouseoff', 'edge', data.eMouseOffEdge = function(e) {
+          if (hoveredGroup) {
+            hoveredGroup = null;
+          }
+        });
+
+        cy.on('mousemove', 'node', data.eMouseMoveNode = function(e) {
+          let node = this;
+          if (!isAGroup(node) && parent.length === 0) {
```

Comment:
What is `parent` supposed to be here? When I debug, it's just the window object.

Reply:
I fixed this up since you're off for a few weeks. Seemed not to need the `parent.length` check, so I took it out. Actually checking if the node has a parent makes hovering over a non-group node while hovering over a group sometimes pick the wrong group. In that case, we want the code to go into the `if` and consider the z depth.